.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+name: GP2GP Build Workflow
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Checkstyle
+        run: |
+          ./gradlew checkStyleMain
+          ./gradlew checkstyleTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Checkstyle Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
+
+  spotbugs:
+    name: Spotbugs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Spotbugs
+        run: |
+          ./gradlew spotbugsMain
+          ./gradlew spotbugsTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
```

Comment:
Before copying I would suggest adding a small defensive check `mkdir -p ./artifacts`,
it will create a directory in case (for any reason) "./artifacts" doesn't exist.
Also, if for any reason the reports don't get copied, could we again be quite explicit and say it:
`cp -r ./service/build/reports ./artifacts || { echo "Copy failed"; exit 1; }`
Again, this is just a defensive approach and I think would be helpful in case of any issues with the pipeline.

Reply:
Hey @ole4ryb - another great point of resiliency! I have added the `-p` flag and structured the workflow a little - as for the second point I think this may be unnecessary as I think the shell is quite descriptive as-is, for example:

`cp: cannot stat './file': No such file or directory`

Reply:
There would be another point of `mv` being more efficient than `cp` - though the performance impact would be non-existent.